### PR TITLE
8254320: Shenandoah: C2 native LRB should activate for non-cset objects

### PR DESCRIPTION
**Pull request by rkennke**
In C2 native-LRB, we don't call the slow-path when the reference is not in cset. However, it may need to return NULL for non-cset objects too, in order to prevent accidental ressurrection of referenced object.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ⏳ (5/5 running) | ⏳ (2/2 running) | ⏳ (2/2 running) |

### Issue
 * [JDK-8254320](https://bugs.openjdk.java.net/browse/JDK-8254320): Shenandoah: C2 native LRB should activate for non-cset objects

### Reviewers
 * [Roland Westrelin](https://openjdk.java.net/census#roland) (@rwestrel - **Reviewer**) ⚠️ Review applies to 05b2bee3a724f1b60fef383c2de8c8520c34c8fb
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/580/head:pull/580`
`$ git checkout pull/580`

### CODE REVIEW

**Review by rwestrel — APPROVED** [review]
Looks good to me.

File: src/hotspot/share/gc/shenandoah/c2/shenandoahSupport.cpp [hunk 1]
```diff
@@ -1342,11 +1342,17 @@ void ShenandoahBarrierC2Support::pin_and_expand(PhaseIdealLoop* phase) {
     // Test for in-cset.
     // Wires !in_cset(obj) to slot 2 of region and phis
     Node* not_cset_ctrl = NULL;
-    test_in_cset(ctrl, not_cset_ctrl, val, raw_mem, phase);
+    if (!lrb->is_native()) {
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/c2/shenandoahSupport.cpp:1345]: I think it deserves a comment why LRB-native is special.

**Review by shipilev — APPROVED** [review]
A minor documentation nit. Otherwise is fine.

File: src/hotspot/share/gc/shenandoah/c2/shenandoahSupport.cpp (hunk 1 above)
Comment (rkennke) [reply src/hotspot/share/gc/shenandoah/c2/shenandoahSupport.cpp:1346]: Yup, I added a comment there. Thanks!

**Review by shipilev — APPROVED** [review]

**Pull request closed without merging**